amlip_docs/rst/demo/tensor_inference.rst

```diff
@@ -0,0 +1,189 @@
+    source $CONDA_PREFIX/etc/conda/activate.d/env_vars.sh
+
+Ensure that you have TensorFlow and TensorFlow Hub installed in your Python environment before proceeding.
+You can install them using pip by executing the following commands:
+
+.. code-block:: bash
+
+    pip3 install tensorflow tensorflow-hub tensorflow-object-detection-api nvidia-cudnn-cu11==8.6.0.163
+
+Additionally, it is required to obtain the TensorFlow model from `TensorFlow Hub <https://tfhub.dev/>`_, follow the steps below:
+
+.. code-block:: bash
+
+    cd ~/AML-IP-ws/src/AML-IP/amlip_demo_nodes/amlip_tensorflow_inference_demo/resource/tensorflow/models/
+    wget -O centernet_hourglass_512x512_kpts_1.tar.gz https://tfhub.dev/tensorflow/centernet/hourglass_512x512_kpts/1?tf-hub-format=compressed
+    mkdir centernet_hourglass_512x512_kpts_1
+    tar -xvf centernet_hourglass_512x512_kpts_1.tar.gz -C ./centernet_hourglass_512x512_kpts_1
+
+Run demo
+========
+
+Run Edge Node
+-------------
+
+In the first terminal, run the following command:
+
+.. code-block:: bash
+
+    # Source colcon installation
+    source install/setup.bash
+
+    # To execute Edge Node to send an image to inferred
+    cd ~/AML-IP-ws/src/amlip/amlip_demo_nodes/amlip_tensorflow_inference_demo/amlip_tensorflow_inference_demo
+    python3 edge_node_async.py
+
+Take into account that this node will wait until there is an *Inference Node* running and available
+in the same :term:`LAN` in order to process the inference.
+The expected output is the following:
+
+.. code-block:: bash
+
+    Edge Node AMLEdgeNode.fb.d4.38.13 ready.
+    Edge Node AMLEdgeNode.fb.d4.38.13 sending data.
+
+    Edge Node received inference from AMLInferenceNode.b8.34.4d.a3
+    Edge Node received inference:
+    Box [(0.15590962767601013, 0.21641747653484344), (0.7388607263565063, 0.7326743006706238)] bicycle: 97%
+    Box [(0.16968876123428345, 0.38129815459251404), (0.403958797454834, 0.9422630071640015)] dog: 92%
+    Box [(0.6158109307289124, 0.13117200136184692), (0.9053990244865417, 0.2978983521461487)] truck: 53%
+    Box [(0.6158109307289124, 0.13117200136184692), (0.9053990244865417, 0.2978983521461487)] car: 48%
+    Box [(0.8892407417297363, 0.19558095932006836), (0.933372974395752, 0.2684069573879242)] potted plant: 34%
+    Box [(0.0753115713596344, 0.15651819109916687), (0.13415342569351196, 0.22736744582653046)] motorcycle: 32%
+
+    Edge Node AMLEdgeNode.fb.d4.38.13 closing.
+
+
+Run Inference Node
+------------------
+
+In the second terminal, run the following command to process the inference:
+
+.. code-block:: bash
+
+    # Source colcon installation
+    source install/setup.bash
+
+    # To execute Inference Node with pre-trained model from TensorFlow
+    cd ~/AML-IP-ws/src/AML-IP/amlip_demo_nodes/amlip_tensorflow_inference_demo/amlip_tensorflow_inference_demo
+    python3 inference_node_async.py
```

Comment:
Following the steps this fails with the following error msg:

```
$ python3 inference_node_async.py
2023-06-22 09:07:57.094502: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2023-06-22 09:07:57.133200: I tensorflow/tsl/cuda/cudart_stub.cc:28] Could not find cuda drivers on your machine, GPU will not be used.
2023-06-22 09:07:57.133590: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-06-22 09:07:57.731403: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
Traceback (most recent call last):
  File "/home/eprosima/projects/current/annapurna/src/amlip/amlip_demo_nodes/amlip_tensorflow_inference_demo/amlip_tensorflow_inference_demo/inference_node_async.py", line 24, in <module>
    from object_detection.utils import label_map_util
  File "/home/eprosima/.local/lib/python3.10/site-packages/object_detection/utils/label_map_util.py", line 21, in <module>
    from object_detection.protos import string_int_label_map_pb2
  File "/home/eprosima/.local/lib/python3.10/site-packages/object_detection/protos/string_int_label_map_pb2.py", line 36, in <module>
    _descriptor.FieldDescriptor(
  File "/home/eprosima/.local/lib/python3.10/site-packages/google/protobuf/descriptor.py", line 561, in __new__
    _message.Message._CheckCalledFromGeneratedFile()
TypeError: Descriptors cannot not be created directly.
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).

More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
```

(fixed calling `pip install protobuf==3.20.*`)